Update repository URL for Atomix

jrhenderson1988 opened the pull request:
[Atomix](https://github.com/atomix/atomix) is currently categorised as Java but it's actually written in Go.

ongardie commented:
Thanks @jrhenderson1988. It looks like the old Atomix has been archived to https://github.com/atomix/atomix-archive. Do you mind forking the entry into two: the old one in Java and the new one in Go?

jrhenderson1988 commented:
Absolutely @ongardie. Happy to fork into two entries. However, after a quick dig into the Go version of Atomix, I'm not actually sure whether the project itself implements Raft or whether it just uses other open source Raft implementations.

For example, see [here](https://github.com/atomix/atomix/blob/e09289f617970cbdf0b1405066e98066dd56cd4e/stores/raft/go.mod#L13) - it looks like that particular module is using Dragonboat. However, [this module](https://github.com/atomix/atomix/tree/master/drivers/raft) doesn't seem to be importing an existing library. To be honest, I'm not familiar enough with the Atomix project to work out whether or not it's a fit. Do you have any thoughts or opinions on this?

ongardie commented:
Looks to me like it's using Dragonboat. https://atomix.io/user-guide/deployment/data-stores/consensus/ also describes it that way. So maybe the right thing to do here is just to point the single JSON entry over to https://github.com/atomix/atomix-archive.

jrhenderson1988 commented:
That makes sense. I've updated the PR to point to the archived repo. I've also updated the name to hopefully make it clear that we mean the archived Java project, rather than the active Go version. Hope that makes sense. Happy to make further changes if you'd like.

ongardie commented:
Merged, thanks for the research and followups @jrhenderson1988
